c

```diff
@@ -0,0 +1,259 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+
+typedef struct radnici {
+	int sifra;
+	char imeRadnika[21];
+	char prezimeRadnika[21];
+	float koeficijenat;
+	int radniStaz;
+}tipRadnika;
+
+
+void unosRadnika(tipRadnika* pradnik) {
+	puts("Unesi podatke o radniku:");
+	printf("Unesi sifru radnika:\n");
+	scanf("%d", &pradnik->sifra);
+	puts("Unesi ime radnika:");
+	scanf("%s", &pradnik->imeRadnika);
+	puts("Unesi prezime radnika:");
+	scanf("%s", &pradnik->prezimeRadnika);
+	puts("Unesi koeficijenat radnika:");
+	scanf("%f", &pradnik->koeficijenat);
+	printf("Unesi radni staz:\n");
+	scanf("%d", &pradnik->radniStaz);
+}
+
+void prikaziRadnika(tipRadnika radnik)
+{
+	puts("\nPodaci o radniku:");
+	printf("Sifra radnika: %d\n", radnik.sifra);
+	printf("Ime radnika: %s\n", radnik.imeRadnika);
+	printf("Prezime radnika: %s\n", radnik.prezimeRadnika);
+	printf("Koeficijenat radnika: %.2f\n", radnik.koeficijenat);
+	printf("Radni staz radnika: %d\n", radnik.radniStaz);
+}
+
+void brisiRadnika(int* n, tipRadnika radnici[], int redniBr) {
+	int i, j = 0;
+	for (i = redniBr - 1; i < *n - 1; i++)
+	{
+		radnici[i] = radnici[i + 1];
+	}
+	(*n)--;
+}
+
+void upisiRadnike(tipRadnika radnik[], int brRadnika) {
+	FILE* fpr = fopen("radnici.txt", "w");
+	if (fpr == NULL) {
+		printf("Greska pri otvaranju fajla\n");
+		return;
+	}
+	for (int i = 0; i < brRadnika; i++) {
+		fprintf(fpr, "%d %s %s %.2f %d\n", radnik[i].sifra, radnik[i].imeRadnika, radnik[i].prezimeRadnika, radnik[i].koeficijenat, radnik[i].radniStaz);
+	}
+	fclose(fpr);
+}
+
+void procitajRadnike(FILE* pfradnik, tipRadnika* radnik)
+{
+	fscanf(pfradnik, "%d", &radnik->sifra);
+	fscanf(pfradnik, "%s", radnik->imeRadnika);
+	fscanf(pfradnik, "%s", radnik->prezimeRadnika);
+	fscanf(pfradnik, "%.2f", &radnik->koeficijenat);
+	fscanf(pfradnik, "%d", &radnik->radniStaz);
+}
+
+void kopiraj_u_fajl(FILE* pftxt, tipRadnika radnik)
+{
+	fprintf(pftxt, "%d\n", radnik.sifra);
+	fprintf(pftxt, "%s\n", radnik.imeRadnika);
+	fprintf(pftxt, "%s\n", radnik.prezimeRadnika);
+	fprintf(pftxt, "%f\n", radnik.koeficijenat);
+	fprintf(pftxt, "%d\n", radnik.radniStaz);
```

Comment:
Unutar fcije `upisiRadnike`, u for petlji mozes da koristis fciju `kopiraj_u_fajl` umesto sto rucno upisujes svakog radnika. 

U for petlji u `upisiRadnike`, ti upisujes podatke svakog radnika u novi red (sto mozda bude teze za citanje). Dok u fciju `kopiraj_u_fajl` ti pises svaki podatak u novi red. Mozda bi hteo da izmenis to da nacin na koji upisujes podatke bude svuda radjen na isti nacin. To ce da spreci buduce greske na koje mozes da naidjes.

Reply:
kopiraj_u_fajl(fpr, radnik[i])

Ovako ?

Reply:
Da, izgledalo bi nesto ovako:

```c
for(int i=0; i<brRadnika; i++){
 kopiraj_u_fajl(fpr, radnici[i]);
}
```